src/java/org/apache/cassandra/service/StorageProxy.java

```diff
@@ -2479,11 +2518,18 @@ public DroppableRunnable(Verb verb)
 
         public final void run()
         {
-            long approxCurrentTimeNanos = MonotonicClock.Global.approxTime.now();
+            approxStartTimeNanos = MonotonicClock.Global.approxTime.now();
```

Comment:
Note: Do we need to make this `volatile` to ensure visibility for the start time for the thread querying the virtual table?